sdk/identity/Azure.Identity/tests/DefaultAzureCredentialTests.cs

```diff
@@ -84,6 +84,40 @@ public void ValidateCtorOptionsPassedToCredentials()
             Assert.AreEqual(expTenantId, actTenantId);
         }
 
+        [Test]
+        [NonParallelizable]
+        public void ValidateEnvironmentBasedOptionsPassedToCredentials([Values]bool clientIdSpecified, [Values]bool usernameSpecified, [Values]bool tenantIdSpecified)
+        {
+            var expClientId = clientIdSpecified ? Guid.NewGuid().ToString() : null;
+            var expUsername = usernameSpecified ? Guid.NewGuid().ToString() : null;
+            var expTenantId = tenantIdSpecified ? Guid.NewGuid().ToString() : null;
+
+            using (new TestEnvVar("AZURE_CLIENT_ID", expClientId))
+            using (new TestEnvVar("AZURE_USERNAME", expUsername))
+            using (new TestEnvVar("AZURE_TENANT_ID", expTenantId))
+            {
+                var credFactory = new MockDefaultAzureCredentialFactory(CredentialPipeline.GetInstance(null));
+
+                credFactory.OnCreateManagedIdentityCredential = (clientId, _) =>
+                {
+                    Assert.AreEqual(expClientId, clientId);
+                };
+
+                credFactory.OnCreateSharedTokenCacheCredential = (tenantId, username, _) =>
+                {
+                    Assert.AreEqual(expTenantId, tenantId);
+                    Assert.AreEqual(expUsername, username);
+                };
+
+                var options = new DefaultAzureCredentialOptions
```

Comment:
What's this for? Did you maybe mean to assert that these `options` match the defined defaults?

Reply:
...i.e. it's not used.

Reply:
yes this was an oversite. updated the test to use the options that are created here.